Runtime/Shapes/ShapeComponent.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+
+namespace UnityEngine.ProBuilder
+{
+    [RequireComponent(typeof(ProBuilderMesh))]
+    public class ShapeComponent : MonoBehaviour
+    {
+        [SerializeReference]
+        public Shape shape = new Cube();
+
+        public bool isInit;
+
+        ProBuilderMesh m_Mesh;
+
+        [SerializeField]
+        Vector3 m_Size;
+
+        [SerializeField]
+        Vector3 m_Rotation;
+
+        [SerializeField]
+        Quaternion m_RotationQuaternion = Quaternion.identity;
+
+        public Quaternion rotationQuaternion {
+            get {
+                return m_RotationQuaternion;
+            }
+            set {
+                m_RotationQuaternion = value;
+                m_Rotation = m_RotationQuaternion.eulerAngles;
+            }
+        }
+
+        public Vector3 size {
+            get { return m_Size; }
+            set { m_Size = value; }
+        } 
+
+        public ProBuilderMesh mesh {
+            get { return m_Mesh == null ? m_Mesh = GetComponent<ProBuilderMesh>() : m_Mesh; }
+        }
+
+        // Bounds where center is in world space, size is mesh.bounds.size
+        internal Bounds meshFilterBounds {
```

Comment:
I have vague memories of this, but can you remind me why we need a bounds that has the center in world space with size in model space?